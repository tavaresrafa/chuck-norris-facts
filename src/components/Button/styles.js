import styled from 'styled-components';
import { colors } from '../../styles/colors';

export const Container = styled.div`
  width: 100%;

  :not(:last-child) {
    margin-right: 0.5rem;
  }

  @media (min-width: 576px) {
    width: auto;
  }
`;

export const Button = styled.button`
  background: ${colors.gray2};
  color: ${colors.black};
  display: flex;
  justify-content: center;
  align-items: center;
  border: 1px solid ${colors.gray2};
  font-size: 1.3rem;
  padding: 1.2rem;
  width: 100%;
  height: 4rem;
  text-align: center;
  cursor: pointer;
  transition: all 0.2s ease-in-out;

  :hover {
    color: ${colors.white};
    background: ${colors.orange};
    border: 1px solid ${colors.orange};
  }
`;
