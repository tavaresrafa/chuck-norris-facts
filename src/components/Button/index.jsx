import React from 'react';
import * as S from './styles';

const Button = ({ text, handleClick }) => (
  <S.Container>
    <S.Button data-testid="button" onClick={handleClick}>{text}</S.Button>
  </S.Container>
)

export default Button;
