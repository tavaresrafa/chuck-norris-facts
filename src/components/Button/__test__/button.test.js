import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../../Button';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button />, div);
});

it('renders button correctly', () => {
    const {getByTestId} = render(<Button text="Search" />);
    expect(getByTestId('button')).toHaveTextContent('Search');
});

// SNAPSHOTS
it('matches snapshot', () => {
    const tree = renderer.create(<Button text="Search" />).toJSON();
    expect(tree).toMatchSnapshot();
});