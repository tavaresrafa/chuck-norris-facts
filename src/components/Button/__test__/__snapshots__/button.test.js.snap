// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-AxjAm bcsFCu"
>
  <button
    className="sc-AxirZ kbCSaR"
    data-testid="button"
  >
    Search
  </button>
</div>
`;
